test/integration/k8s-deploy-test.py

```diff
@@ -124,10 +123,9 @@ def verifyService(service, parsedArgs):
             return dictMatch, msg
 
     if annotationsKey in parsedArgs:
-        dictMatch, msg = compareDicts(
-            service['metadata']['annotations'], parsedArgs[annotationsKey], annotationsKey)
-        if not dictMatch:
-            return dictMatch, msg
+        for key in parsedArgs[annotationsKey]:
+            if key not in service['metadata']['annotations'].keys():
+                return False, f"expected annotation {key} not found in service annotations. \n actual annotation keys {','.join(service['metadata']['annotations'].keys())}"
```

Comment:
@jaiveerk I tried to explain in https://github.com/Azure/k8s-deploy/pull/287#issuecomment-1671161746

> Since annotations in general consist of more than just resource annotations, for example `deployment.kubernetes.io/revision` and `kubectl.kubernetes.io/last-applied-configuration`, it was not sufficient to check whether the annotation is not present when not annotating the resources. Some standard annotations will always be there.
> 
> Also, testing the annotations in the integration test on key value following this format (`testkey1:testvalue1,testkey2:testvalue2`) is hard to realise since annotations, especially for child annotations, can contain full dumps of the objects. Objects with variable content are hard to test this way. That's why I choose to test on keys only following the format `testkey1,testkey2,testkey3`.

Reply:
ah I see - apologies for forgetting about that! would you mind mofidying the compareDicts function accordingly, or at least abstracting out the repeated logic between verifyDeployment and verifyService to a single function like compareAnnotations?

Reply:
@jaiveerk I just abstracted out the logic by implementing the method `validateKeyPresence`